### ISSUE TITLE
S3 Async Client with AWS-CRT downloading S3 getObject file without kms decrypt permissions

### ISSUE THREAD

**Issue by bharathjagini** [aws/aws-sdk-java-v2#4684]
### Describe the bug

S3 async client with AWS CRT downloading the S3 bucket files having KMS key encryption even though kms decrypt permission is not given.

### Expected Behavior

It should not download and throw exception/error when tried to download using getObject.

### Current Behavior

Its downloading the object irrespective of whether kms key have permissions to decrypt for the principal/role. I can see the error for kms:decrypt permissions issue in the content of the file.

### Reproduction Steps

1. Create bucket with kms key encryption but valid S3 bucket permissions like GetObject to the principal.
2. Don't give kms decrypt action for the principal in the kms key permissions
3. Use S3 async Client with Aws-crt to download the file using getObject

### AWS Java SDK version used

2.0

### JDK version used

11

### Operating System and version

linux

**Comment by bharathjagini** [comment]
To give more insights,I tried to reproduce the issue again, error code and error information is getting added to the downloaded file with xml error content and did not throw any exceptions regarding the same.

**Comment by bhoradc** [comment]
Hi @bharathjagini,

Thank you for reporting this issue. I am able to reproduce the above mentioned scenario. 

Below are the steps for the same. 

1. Sample code using AWS SDK for Java version 2.21.23.
```java
public class Crt4684 {
    public static void main(String[] args) {

            //AWS CRT-based S3 client
            S3AsyncClient s3Client = S3AsyncClient.crtCreate();

            // Download an object from Amazon S3 to a local file.
            GetObjectRequest objectRequest = GetObjectRequest
                    .builder()
                    .key("keyname")
                    .bucket("bucketname")
                    .build();
            Path path = Path.of("filename");

            CompletableFuture<GetObjectResponse> objectBytes = s3Client.getObject(objectRequest, path);

            objectBytes.whenComplete((response,exception) -> {
                if (response == null)
                {
                    System.out.println("Downloaded sucessfully at " + path);
                }
                else
                {
                    exception.printStackTrace();
                }
            });
        objectBytes.join();
        }
}
``` 
2. Updated bucket policy to allow GetObject().
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Statement1",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::<<redacted>>:user/User"
            },
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3:::bucketname/*"
        }
    ]
}
```
3. Updated the Bucket Encryption Type to SSE-KMS.
4. KMS key used doesn't have `KMS:Decrypt` permission for the User. 
```json
            "Sid": "Allow use of the key",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::<<redacted>>:user/User"
            },
            "Action": [
                "kms:Encrypt"
            ],
            "Resource": "*"
```
5. With no KMS:Decrypt permission, SDK is downloading the object file with AccessDenied error for `S3AsyncClient CRT*builder (S3AsyncClient.crtCreate(), S3AsyncClient.crtbuilder())`. 
6. This behaviour isn't consistent with standard builder(S3AsyncClient.builder()), where the AccessDenied exception is thrown without downloading the object file. 

We'll review this with the team and provide further updates. 

Regards, 
Chaitanya

**Comment by debora-ito** [comment]
@bharathjagini 

A fix was released as part of version `2.22.4`, please try it out and let us know if you have any other issue.
